migrations/20221114135330_add_email_column_identities.up.sql

```diff
@@ -0,0 +1,6 @@
+alter table only {{ index .Options "Namespace" }}.identities
+  add column if not exists email text generated always as (lower(identity_data->>'email')) stored;
+
+comment on column {{ index .Options "Namespace" }}.identities.email is 'Auth: Email is a generated column that references the optional email property in the identity_data';
+
+create unique index if not exists identities_email_provider_email_key on {{ index .Options "Namespace" }}.identities (email) where (provider = 'email');
```

Comment:
hmm will creating a unique index fail if `auth.identities.email` is null? we only recently added the change to include the `identity_data->>'email'` claim for those with `provider = email` so there could be some users with `identity_data->>'email' = null` 

will take a look at backfilling this data from the `auth.users.email` column and benchmark the backfilling process

Reply:
> will creating a unique index fail if auth.identities.email is null?

Nulls are distinct by nature, so shouldn't be a problem.

Reply:
Added a query that should run in `O(n log(n))` time... I don't think it's possible to achieve anything quicker than this given that you still need to scan most of the rows with an `email` provider, and then do a `log(n)` lookup on the user ID.

Reply:
I'm not using a `join` clause because of this statement from the [PostgreSQL docs](https://www.postgresql.org/docs/current/sql-update.html):

> When a FROM clause is present, what essentially happens is that the target table is joined to the tables mentioned in the from_item list, and each output row of the join represents an update operation for the target table. When using FROM you should ensure that the join produces at most one output row for each row to be modified. In other words, a target row shouldn't join to more than one row from the other table(s). If it does, then only one of the join rows will be used to update the target row, but which one will be used is not readily predictable.
>
> Because of this indeterminacy, referencing other tables only within sub-selects is safer, though often harder to read and slower than using a join